frontend/pages/queries/ManageQueriesPage/ManageQueriesPage.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useState, useCallback, useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import { push } from "react-router-redux";
+import { IQuery } from "interfaces/query";
+import { IUser } from "interfaces/user";
+
+// @ts-ignore
+import queryActions from "redux/nodes/entities/queries/actions";
+// @ts-ignore
+import { renderFlash } from "redux/nodes/notifications/actions";
+
+import paths from "router/paths";
+import permissionUtils from "utilities/permissions";
+
+import Button from "components/buttons/Button";
+import QueriesListError from "./components/QueriesListError";
+import QueriesListWrapper from "./components/QueriesListWrapper";
+import RemoveQueryModal from "./components/RemoveQueryModal";
+
+const baseClass = "manage-queries-page";
+interface IRootState {
+  auth: {
+    user: IUser;
+  };
+  entities: {
+    queries: {
+      isLoading: boolean;
+      data: IQuery[];
+      errors: any;
+    };
+  };
+}
+
+const renderTable = (
+  onRemoveQueryClick: React.MouseEventHandler<HTMLButtonElement>,
+  queriesList: IQuery[],
+  queriesErrors: any
+): JSX.Element => {
+  if (Object.keys(queriesErrors).length !== 0) {
+    return <QueriesListError />;
+  }
+
+  return (
+    <QueriesListWrapper
+      onRemoveQueryClick={onRemoveQueryClick}
+      queriesList={queriesList}
+    />
+  );
+};
+
+const ManageQueriesPage = (): JSX.Element => {
+  const currentUser = useSelector((state: IRootState) => state.auth.user);
+  const isOnlyObserver = permissionUtils.isOnlyObserver(currentUser);
+
+  const dispatch = useDispatch();
+  const { NEW_QUERY } = paths;
+  const onCreateQueryClick = () => dispatch(push(NEW_QUERY));
+
+  useEffect(() => {
+    dispatch(queryActions.loadAll());
+  }, [dispatch]);
+
+  const queries = useSelector((state: IRootState) => state.entities.queries);
+  const queriesList = Object.values(queries.data);
+  const queriesErrors = queries.errors;
+
+  const [selectedQueryIds, setSelectedQueryIds] = useState([]);
+  const [showRemoveQueryModal, setShowRemoveQueryModal] = useState(false);
```

Comment:
Let's start typing the states because it will provide better visibility further in the code. Based on what I see, I think the first state would be `const [selectedQueryIds, setSelectedQueryIds] = useState<IQuery[]>([]);`